README.md
# About this repository
I'm not going to say much because the in-depth info is avaliable [here (in this repository's wiki)](https://github.com/CraftRBLX/CraftRBLX.github.io/wiki).

1. This repository is hosted at https://craftrblx.github.io
2. This was mainly created around my [main YouTube channel](https://www.youtube.com/c/CraftRBLX)

![Your Repository's Stats](https://github-readme-stats.vercel.app/api?username=CraftRBLX&show_icons=true)

![Your Repository's Stats](https://github-readme-stats.vercel.app/api/top-langs/?username=CraftRBLX&theme=blue-green)

### OTHER THINGS:
Website Homepage: https://craftrblx.github.io<br>
Website Changelog: https://craftrblx.github.io/changelog.html<br>
Website Changelog (On GitHub): https://github.com/CraftRBLX/CraftRBLX.github.io/blob/master/CHANGELOG.md<br>

### Contact
My Main Youtube channel: [CraftRBLX](https://www.youtube.com/c/CraftRBLX)<br>
My Coding YouTube Channel: [Code Vibes](https://www.youtube.com/channel/UC2OUBrGSPqJBm4A32YVbe9A)<br>
My Discord Server: [The CraftRBLX Official Fan Discord Server](https://discord.gg/BHhV8fH)<br>
My Roblox Group: [CraftRBLX Official Fan Roblox Group](https://www.roblox.com/groups/10295625/CraftRBLX-Official-Fan-Roblox-Group#!/about)<br>
Website: [The CraftRBLX Official Website](https://craftrblx.github.io)<br>
Email: Not Public<br>
Phone Number: Non-existant


CHANGELOG.md
# Change log
## Main = Default English, FE = Funny English, ES = Spanish

## MAIN v8.1 - Jun 23, 2021
**Changes**
- The Page Loader is now back on all pages except the (homepage)[https://craftrblx.github.io]
- Updated one of the Easter Eggs

**Added**
- Added a new typewriter/typing effect on the (homepage)[https://craftrblx.github.io]

**Removed**
- Removed 1 Easter Egg

Easter Egg Total: 2

---

## MAIN v8.0 - Jun 1, 2021
**Changes**
- Light Mode has now been scrapped, the site will now permanently use dark mode.
- The Light and Dark mode buttons have been removed to give the site a cleaner look (also because the site is now permanently on dark mode).
- The website version has been removed from most pages except the (homepage)[https://craftrblx.github.io] and the (changelog)[https://craftrblx.github.io/changelog.html#version-area].

---

## MAIN v7.7 - May 3, 2021
**Changes**
- 100% remade the [Error 404 page](https:/craftrblx.github.io/404.html), to be more user-friendly
- Changed the old CraftRBLX logo on the nav to the new one

---

## MAIN v7.6 - May 1, 2021
**Added**
- Added a home SVG next to the homepage text in the pages part of the nav.

**Changes**
- Made it so that if you load the website without Javascript, you'll get a warning message. Instead of the page loader spinning forever.
- Changed the favicon of all pages to the [new CraftRBLX logo](https://craftrblx.github.io/assets/logos/crlogo2021S1.png).

---

## MAIN v7.5 - Apr 29, 2021
**Added**
- Made it so that when you hover over a link in the blog section of the [homepage](https://craftrblx.github.io/), the underline appears. Otherwise, it's hidden.
- Added a "[Read Full Blog...]" link to some of the blogs. This link will take you to a page with the full details & content of the blog.

---

## MAIN v7.4 (Easter egg update 2) - Apr 21, 2021
**Added**
- Added another easter egg

**Changes**
- Changed all of the easter eggs to include more stuff in them

Easter Egg Total: 3

---

## MAIN v7.3 (Easter egg update 1) - Apr 20, 2021
**Added**
- Added a "Beta" tag/label to the blogs section on the [homepage](https://craftrblx.github.io/)
- Added a blog posts number (basically a number that tells you how many blog posts there are) to the blogs section on the [homepage](https://craftrblx.github.io/)
- Added a new easter egg 🥚🥚 to the blogs section on the [homepage](https://craftrblx.github.io/)!

**Changes**
- Changed the favicon of all the pages to have a transparent background

Easter Egg Total: 2

---

## MAIN v7.2 - Apr 17, 2021
**Added**
- Added blogs to the [homepage](https://craftrblx.github.io/)

---

## MAIN v7.1 - Apr 12, 2021
**Added**
- Made [allies page](https://craftrblx.github.io/allies.html)
- Published [allies page](https://craftrblx.github.io/allies.html)
- Made [allies page](https://craftrblx.github.io/allies.html) public in the "[CraftRBLX](https://www.youtube.com/channel/UCEh6ZxENe-OzEu_tl3t3s-w)" section of the nav bar

---

## MAIN v7.0 - Apr 8, 2021
**Added**
- Added "[CraftRBLX](https://www.youtube.com/channel/UCEh6ZxENe-OzEu_tl3t3s-w?)" section to nav bar
- Added "[Support CraftRBLX](https://craftrblx.github.io/support.html)" page

**Changes**
- Moved the "[All videos](https://www.youtube.com/channel/UCEh6ZxENe-OzEu_tl3t3s-w/videos)" link from the "videos" section in the nav bar to the "[CraftRBLX](https://www.youtube.com/channel/UCEh6ZxENe-OzEu_tl3t3s-w?)" section in the nav bar
- Moved the [logo](https://craftrblx.github.io/rana_crafrblx.png) back into the correct place

---

## MAIN v6.9 - Mar 16, 2021
**Changes**
- Changed the dark mode version of the light and dark buttons (for safari compatibility).

---

## MAIN v6.8 - Mar 13, 2021
**Added**
- Added some CSS Media queries to adjust the look of the [videos page](https://craftrblx.github.io/videos.html) on Phone/iPhone and Tablet/iPad view.

---

## MAIN v6.7 - Mar 12, 2021
**Changes**
- Changed it so that the loading screen animation, doesn't stop at 0.6 seconds instead its length depends on how long the page takes to load (which in turn depends on how fast your Wi-Fi/Ethernet/Hotspot connection is).

---

## MAIN v6.6 - Mar 11, 2021
**Added**
- Added "TOP 5 MOST POPULAR [CRAFTRBLX VIDEOS](https://www.youtube.com/channel/UCEh6ZxENe-OzEu_tl3t3s-w/videos)" section to the [videos page](https://craftrblx.github.io/videos.html).
- Added "ALL [CRAFTRBLX PLAYLISTS](https://www.youtube.com/channel/UCEh6ZxENe-OzEu_tl3t3s-w/playlists)" section to the [videos page](https://craftrblx.github.io/videos.html).

**Removed**
- The one carousel from the [videos page](https://craftrblx.github.io/videos.html).

**Changes**
- Made all links on all pages (except the path links) the rgb(41,101,213) color.

---

## MAIN v6.5 - Mar 11, 2021
**Announcements**
- THE [VIDEOS PAGE](https://craftrblx.github.io/videos.html) IS NOW OFFICIALLY FINISHED REMODELING

**Added**
- Added [videos page](https://craftrblx.github.io/videos.html) to navigator
- Added a video carousel to the [videos page](https://craftrblx.github.io/videos.html)
- Made all the website text unhighlightable

**Removed**
- Removed "[Most Popular video](https://www.youtube.com/watch?v=pS3GVPsTW58)" section from the [videos page](https://craftrblx.github.io/videos.html).
- Removed "[MCRAFT Season 1 episode 1](https://www.youtube.com/watch?v=gqik66poDWw)" video from the [videos page](https://craftrblx.github.io/videos.html).

---

## MAIN v6.4 - Mar 10, 2021
**Added**
- Added [README.md](https://craftrblx.github.io/README.md) to this website's GitHub repository.
- Added [3 GitHub Wiki pages](https://github.com/CraftRBLX/CraftRBLX.github.io/wiki) to this website's GitHub repository.

---

## MAIN v6.3 - Mar 10, 2021
**Changes**
- Changed the [cookie use policy](https://craftrblx.github.io/cookies.html). SO PLEASE REREAD THEM.

---

## MAIN: v6.2 - Feb 26, 2021
**Changes**
- Changed the color of all links (except the path links) to the royal blue color.

---

## MAIN v6.1 - Feb 26, 2021
**Added**

toggle.js
// function to toggle dark mode
// turn on if newState is "on", otherwise, turn off dark mode
function toggleDarkMode(newState) {
    if (newState == "on") {
        // enable dark mode using the DarkReader API
        DarkReader.enable({ contrast: 150 });
        // set the toggle button to have the sun icon
        document.querySelector("div.dark-mode-toggle").firstElementChild.className = "gg-sun";
        // insert style element to apply some minor adjustments
        setCookie("darkmode", "on", 9999);
    }
    // turning off dark mode
    else {
        // disable dark mode using the DarkReader API
        DarkReader.disable();
        // set the toggle button to have the moon icon
        document.querySelector("div.dark-mode-toggle").firstElementChild.className = "gg-moon";
        // find the style element with the modifications
        setCookie("darkmode", "off", 9999);
    }
}

// add an event listener to detect clicking on the dark mode toggle button
document.querySelector("div.dark-mode-toggle").addEventListener("click", function () {
    // if there is an element with the class "darkreader" on the page (DarkReader is enabled)
    if (document.querySelector(".darkreader")) {
        // turn off dark mode
        toggleDarkMode("off");
    }
    else {
        // turn on dark mode
        toggleDarkMode("on");
    }
}, false);

// code from w3schools to set a cookie (https://www.w3schools.com/js/js_cookies.asp)
// parameters: name of the cookie, value to set, number of days until it expires
function setCookie(cname, cvalue, exdays) {
    var d = new Date();
    d.setTime(d.getTime() + (exdays * 24 * 60 * 60 * 1000));
    var expires = "expires=" + d.toUTCString();
    document.cookie = cname + "=" + cvalue + ";" + expires + ";path=/";
}

// code from w3schools to get the value of a cookie (https://www.w3schools.com/js/js_cookies.asp)
// parameters: name of the cookie
function getCookie(cname) {
    var name = cname + "=";
    var ca = document.cookie.split(';');
    for (var i = 0; i < ca.length; i++) {
        var c = ca[i];
        while (c.charAt(0) == ' ') {
            c = c.substring(1);
        }
        if (c.indexOf(name) == 0) {
            return c.substring(name.length, c.length);
        }
    }
    return "";
}

// when the page loads, toggle dark mode according to the cookie settings
window.addEventListener("load", function () {
    // get the dark mode cookie value
    var darkModeCookie = getCookie("darkmode");
    // if the cookie is set to "on", turn on dark mode
    if (darkModeCookie == "on") {
        // toggle dark mode
        toggleDarkMode(darkModeCookie);
    }
}, false);

dark-mode-switch.js
(function() {
  var darkSwitch = document.getElementById("darkSwitch");
  if (darkSwitch) {
    initTheme();
    darkSwitch.addEventListener("change", function(event) {
      resetTheme();
    });
    function initTheme() {
      var darkThemeSelected =
        localStorage.getItem("darkSwitch") !== null &&
        localStorage.getItem("darkSwitch") === "dark";
      darkSwitch.checked = darkThemeSelected;
      darkThemeSelected
        ? document.body.setAttribute("data-theme", "dark")
        : document.body.removeAttribute("data-theme");
    }
    function resetTheme() {
      if (darkSwitch.checked) {
        document.body.setAttribute("data-theme", "dark");
        document.getElementsByClassName("subscribe")[0].style.color = "black";
        localStorage.setItem("darkSwitch", "dark");
      } else {
        document.body.removeAttribute("data-theme");
        localStorage.removeItem("darkSwitch");
      }
    }
  }
})();


cookieconsentpopup.js
function Pop(){var cssRuleFile="https://cookieconsent.popupsmart.com/src/css/style.css";let lnk=document.createElement('link');lnk.setAttribute('rel','stylesheet');lnk.setAttribute('type','text/css');lnk.setAttribute("href",cssRuleFile);document.getElementsByTagName('head')[0].appendChild(lnk);let styl='undefined';var conDivObj;var fadeInTime=10;var fadeOutTime=10;let cookie={name:'cookieconsent_status',path:'/',expiryDays:365*24*60*60*5000,};let content={message:'This website uses cookies to ensure you get the best experience on our website.',btnText:'Got it!',mode:'  banner bottom',theme:' theme-classic',palette:' palette1',link:'Learn more',href:'https://www.cookiesandyou.com',target:'_blank',};let createPopUp=function(){console.log(content);if(typeof conDivObj==='undefined'){conDivObj=document.createElement('DIV');conDivObj.style.opacity=0;conDivObj.setAttribute('id','spopupCont');}
conDivObj.innerHTML='<div id="poper" class="window '+
content.mode+
content.theme+
content.palette+
'"><span id="msg" class="message">'+
content.message+
'<a id="plcy-lnk" class="policylink" href="'+
content.href+
'"'+
' target='+
content.target+
'>'+
content.link+
'</a></span><span class="credit"><a href="https://craftrblx.github.io/homepage.html" target="_blank">Powered by CraftRBLX</a></span><div id="btn" class="compliance"><a href="#" id="cookie-btn" class="spopupbtnok" >'+
content.btnText+
'</a></div></div>';document.body.appendChild(conDivObj);fadeIn(conDivObj);document.getElementById('cookie-btn').addEventListener('click',function(){saveCookie();fadeOut(conDivObj);});};let fadeOut=function(element){var op=1;var timer=setInterval(function(){if(op<=0.1){clearInterval(timer);conDivObj.parentElement.removeChild(conDivObj);}
element.style.opacity=op;element.style.filter='alpha(opacity='+op*100+')';op-=op*0.1;},fadeOutTime);};let fadeIn=function(element){var op=0.1;var timer=setInterval(function(){if(op>=1){clearInterval(timer);}
element.style.opacity=op;element.style.filter='alpha(opacity='+op*100+')';op+=op*0.1;},fadeInTime);};let checkCookie=function(key){var keyValue=document.cookie.match('(^|;) ?'+key+'=([^;]*)(;|$)');return keyValue?true:false;};let saveCookie=function(){var expires=new Date();expires.setTime(expires.getTime()+cookie.expiryDays);document.cookie=cookie.name+
'='+
'ok'+
';expires='+
expires.toUTCString()+
'path='+
cookie.path;};this.init=function(param){if(checkCookie(cookie.name))return;if(typeof param==='object'){if('ButtonText'in param)content.btnText=param.ButtonText;if('Mode'in param)content.mode=' '+param.Mode;if('Theme'in param)content.theme=' '+param.Theme;if('Palette'in param)content.palette=' '+param.Palette;if('Message'in param)content.message=param.Message;if('LinkText'in param)content.link=param.LinkText;if('Location'in param)content.href=param.Location;if('Target'in param)content.target=param.Target;if('Time'in param)
setTimeout(function(){createPopUp();},param.Time*1000);else createPopUp();}};}
window.start=new Pop();
